src/sentry/utils/registry.py

```diff
@@ -0,0 +1,35 @@
+from __future__ import annotations
+
+from typing import Generic, TypeVar
+
+
+class AlreadyRegisteredError(ValueError):
+    pass
+
+
+class NoRegistrationExistsError(ValueError):
+    pass
+
+
+T = TypeVar("T")
+
+
+class Registry(Generic[T]):
+    def __init__(self):
+        self.registrations: dict[str, T] = {}
+
+    def register(self, key: str):
+        def inner(item: T) -> T:
+            if key in self.registrations:
+                raise AlreadyRegisteredError(
+                    f"A registration already exists for {key}: {self.registrations[key]}"
+                )
+            self.registrations[key] = item
+            return item
+
+        return inner
+
+    def get_registration(self, key: str) -> T:
```

Comment:
nit: since this will be used widely throughout the codebase, would make sense to just call this just get()? That way, the code reads a bit more idiomatic to me: registry.get("my key")

Reply:
Yeah, I was thinking about this... `registry.get` is clear and concise, since `registry.get_registration` is repeating registry. I'll change it